src/ray/raylet/scheduling/cluster_resource_data.cc

```diff
@@ -538,6 +538,29 @@ bool ResourceRequest::IsEmpty() const {
   return true;
 }
 
+bool ResourceRequest::operator==(const ResourceRequest &other) const {
```

Comment:
I am not sure if `requires_object_store_memory` should be take into account. cc @stephanie-wang  cc @scv119

Reply:
Comparing this field gives a more complete `==` operation. Anyway, I've added this for the time being, and we can just drop it if it makes no sense but only overhead.

Reply:
yeah it depends on where it used.
if it's for representing resources of a node, it's not useful; but for presenting a resource request, we should take it into consideration.

that's exactly why we need two classes (resource set and resource request)